### DISCUSSION TITLE
react-markdown: `allowDangerousHtml` not working (but `escapeHtml` does work)

### DISCUSSION THREAD

**Discussion by RDjarbeng** [remarkjs/remark#551] [Q&A] [answered]
# React markdown HTML not parsing workaround

`import ReactMarkdownWithHtml from "react-markdown/with-html"
import myobject from 'contentJSON/homeInvestors.json'
`
Contents of `myobject`
`myobject ={content='Welcome <mark className ="bg-warning"> To the community</mark>', ...}`

I try to add the content property (as given in the documentation) like so:
`<ReactMarkdownWithHtml children={myobject.content} allowDangerousHtml />`
but it did not parse the HTML.
Then I changed it to 
`<ReactMarkdownWithHtml children={myobject.content} allowDangerousHtml={true} escapeHtml={false} />`
and it worked fine.

**Is this a safe or a recommended way?**

 The documentation does not mention 'escapeHtml'. Why does the example from the documentation not work for me, is it because of the className attribute? 
I would like to know just in case this work around fails one day. Also just so anybody looking for a solution can find it here

**Comment by wooorm** [comment, marked as answer]
Please provide information on what versions of package you are using.

See this [cheatsheet](https://commonmark.org/help/) on markdown. Specifically: using fenced code here is useful so you can share more of your code.

**Reply by ChristianMurphy** [reply]
@wooorm I'm able to reproduce this on 4.3.1 https://codesandbox.io/s/condescending-darkness-u6mk1

**Reply by wooorm** [reply]
Right. We’re on 5.0.0+ now, which added `allowDangerousHtml`, in favor of `escapeHtml`: https://github.com/remarkjs/react-markdown/issues/484

**Reply by RDjarbeng** [reply]
In reply to [wooorm](https://github.com/wooorm)
From my package.json:
`"react-markdown": "^4.3.1"`
And I did an npm install about 3 days ago

**Reply by ChristianMurphy** [reply]
@RDjarbeng, @wooorm is saying that in v4 `escapeHtml={false}` is used, in v5 `allowDangerousHtml` is used.
Trying to use v5 attributes in v4 won't work:

For example:
* v4 https://codesandbox.io/s/elastic-blackwell-9qtsz
* v5 https://codesandbox.io/s/vigilant-bogdan-6d3ei

**Reply by wooorm** [reply]
What Christian said indeed, and for an actionable answer:

Upgrade `react-markdown` to use `allowDangerousHtml` (specifically, manually update it in your `package.json` to `^5.0.0`, and then `npm install` again)

**Reply by RDjarbeng** [reply]
> @RDjarbeng, @wooorm is saying that in v4 `escapeHtml={false}` is used, in v5 `allowDangerousHtml` is used.
Ok thanks @ChristianMurphy 

Would like to know how to always use the  latest version because I installed it by following the instructions on the main branch.
`npm install react-markdown`
I'm surprised to find I received an older version
Thanks for your assistance

**Reply by RDjarbeng** [reply]
> Upgrade `react-markdown` to use `allowDangerousHtml` (specifically, manually update it in your `package.json` to `^5.0.0`, and then `npm install` again)

Thanks @wooorm your answer came before my question
I guess this issue is resolved

**Reply by ChristianMurphy** [reply]
`npm install react-markdown` will give the latest version, if you didn't recieve it, you likely ran `npm install react-markdown` before the release was available on npm, see https://www.npmjs.com/package/react-markdown (versions) for a past release timeline.

**Reply by wooorm** [reply]
@RDjarbeng See also https://softwareengineering.stackexchange.com/questions/331648/a-good-node-js-workflow-for-keeping-packages-up-to-date on how to keep your deps up to date!

Personally, I use https://github.com/raineorshine/npm-check-updates to check for updates locally
